jetbrains-core/tst/software/aws/toolkits/jetbrains/core/AwsResourceCacheTest.kt

```diff
@@ -475,6 +480,50 @@ class AwsResourceCacheTest {
         assertThat(sut.getResourceIfPresent(mockResource, connectionSettings)).isEqualTo("hello")
     }
 
+    @Test
+    fun concurrentlyRunningExceptionalResourcesGetTheSameException() {
+        val latch = CountDownLatch(1)
+        whenever(mockResource.fetch(any(), any())).then {
+            latch.await(500, TimeUnit.MILLISECONDS)
+            throw RuntimeException("Boom")
+        }
+
+        val first = sut.getResource(mockResource, connectionSettings)
+        val second = sut.getResource(mockResource, connectionSettings)
+        latch.countDown()
+        assertThat(first).wait().hasException.withFailMessage { "" }
```

Comment:
assert that it's the same exception and not 2 different calls?

Reply:
added a verify that the mock only gets hit once